File: Leetcode/islandPerimeter.cpp
class Solution {
public:
    int islandPerimeter(vector<vector<int>>& grid) {
        int row = grid.size();
        int col = grid[0].size();
        int num = 0;
        for(int i = 0; i < row; i++){
            for(int j = 0; j < col; j++){
                if(grid[i][j] == 1){
                    if(i == 0 || grid[i-1][j] == 0) num++; //up
                    if(j == 0 || grid[i][j-1] == 0) num++; //left
                    if(i == row - 1 || grid[i + 1][j] == 0) num++; //down
                    if(j == col - 1 || grid[i][j + 1] == 0) num++;//right
                }
            }
        }
        return num;
    }
};

File: Leetcode/islandPerimeter.java
class Solution {
  public int islandPerimeter(int[][] grid) {
    int row = grid.length;
    int col = grid[0].length;
    // cari yang land
    for (int i = 0; i < row; i++) {
      for (int j = 0; j < col; j++) {
        if (grid[i][j] == 1)
          return countPerimeter(grid, i, j);
      }
    }
    return -1;
  }

  public int countPerimeter(int[][] grid, int row, int col) {
    if (row < 0 || row >= grid.length || col < 0 || col >= grid[0].length || grid[row][col] == 0)
      return 1;

    // mark as visited
    if (grid[row][col] == -1)
      return 0;
    grid[row][col] = -1;

    int count = 0;

    count += countPerimeter(grid, row - 1, col);
    count += countPerimeter(grid, row + 1, col);
    count += countPerimeter(grid, row, col - 1);
    count += countPerimeter(grid, row, col + 1);

    return count;
  }
}